feature/experimental/gribi/ate_tests/supervisor_failure/Supervisor_Failure_test.go

```diff
@@ -0,0 +1,364 @@
+		FibACK:               false,
+		Persistence:          true,
+		InitialElectionIDLow: 10,
+	}
+	defer clientA.Close(t)
+	if err := clientA.Start(t); err != nil {
+		t.Fatalf("gRIBI Connection can not be established")
+	}
+
+	args := &testArgs{
+		ctx:     ctx,
+		clientA: &clientA,
+		dut:     dut,
+		ate:     ate,
+		top:     top,
+	}
+	// Program a route and ensure AFT telemetry returns FIB_PROGRAMMED
+	routeInstall(ctx, t, args)
+	// Verify that static route(203.0.113.0/24) to ATE port-2 is preferred by the traffic.`
+	srcEndPoint := args.top.Interfaces()[atePort1.Name]
+	dstEndPoint := args.top.Interfaces()[atePort2.Name]
+	flow := testTraffic(t, args.ate, args.top, srcEndPoint, dstEndPoint)
+
+	controllers := cmp.FindComponentsByType(t, dut, controlcardType)
+	t.Logf("Found controller list: %v", controllers)
+	// Only perform the switchover for the chassis with dual controllers.
+	if len(controllers) != 2 {
+		t.Skipf("Dual controllers required on %v: got %v, want 2", dut.Model(), len(controllers))
+	}
+
+	secondaryBeforeSwitch, primaryBeforeSwitch := findSecondaryController(t, dut, controllers)
+	t.Logf("Detected Secondary: %v, Primary: %v", secondaryBeforeSwitch, primaryBeforeSwitch)
+
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	switchoverRequest := &spb.SwitchControlProcessorRequest{
+		ControlProcessor: &tpb.Path{
+			Elem: []*tpb.PathElem{{Name: secondaryBeforeSwitch}},
+		},
+	}
+	t.Logf("switchoverRequest: %v", switchoverRequest)
+	switchoverResponse, err := gnoiClient.System().SwitchControlProcessor(context.Background(), switchoverRequest)
+	if err != nil {
+		t.Fatalf("Failed to perform control processor switchover with unexpected err: %v", err)
+	}
+	t.Logf("gnoiClient.System().SwitchControlProcessor() response: %v, err: %v", switchoverResponse, err)
+
+	secondaryAfterSwitch, primaryAfterSwitch := findSecondaryController(t, dut, controllers)
+	t.Logf("Found Secondary Controller after switchover: %v, Primary: %v", secondaryAfterSwitch, primaryAfterSwitch)
+
+	startSwitchover := time.Now()
+	t.Logf("Wait for new Primary controller to boot up by polling the telemetry output.")
+	for {
+		var currentTime string
+		t.Logf("Time elapsed %.2f seconds since switchover started.", time.Since(startSwitchover).Seconds())
+		time.Sleep(30 * time.Second)
+		if errMsg := testt.CaptureFatal(t, func(t testing.TB) {
+			currentTime = dut.Telemetry().System().CurrentDatetime().Get(t)
+		}); errMsg != nil {
+			t.Logf("Got testt.CaptureFatal errMsg: %s, keep polling ...", *errMsg)
+		} else {
+			t.Logf("Controller switchover has completed successfully with received time: %v", currentTime)
+			break
+		}
+		if uint64(time.Since(startSwitchover).Seconds()) > maxSwitchoverTime {
+			t.Fatalf("time.Since(startSwitchover): got %v, want < %v", time.Since(startSwitchover), maxSwitchoverTime)
+		}
+	}
+	t.Logf("Controller switchover time: %.2f seconds", time.Since(startSwitchover).Seconds())
+
+	t.Log("Validate OC Switchover time/reason.")
```

Comment:
Can we move all the `Telemetry` validations to a separate function and call it from here?

Reply:
Implemented.. New code push failed due to "Needs /gcbrun from a collaborator"..